src/supautils.c

```diff
@@ -448,29 +449,36 @@ static void supautils_hook(PROCESS_UTILITY_PARAMS) {
 
       constrain_extension(stmt->extname, cexts, total_cexts);
 
-      if (is_extension_privileged(stmt->extname, privileged_extensions)) {
-         bool already_switched_to_superuser = false;
+      bool already_switched_to_superuser = false;
 
-         switch_to_superuser(supautils_superuser, &already_switched_to_superuser);
+      switch_to_superuser(supautils_superuser, &already_switched_to_superuser);
```

Comment:
Ditto. I find these `already_switched_to_superuser` flags mysterious. When can this happen? Is it tested? If not, please mention how can a test be added to cover this behavior.